core/src/test/java/com/coderanch/blackjack/HandTest.java

```diff
@@ -0,0 +1,173 @@
+            List.of(
+                    new HandTestArgument(
+                            List.of(new Card(Rank.ACE, Suit.CLUBS),
+                                    new Card(Rank.ACE, Suit.CLUBS),
+                                    new Card(Rank.KING, Suit.CLUBS),
+                                    new Card(Rank.EIGHT, Suit.CLUBS),
+                                    new Card(Rank.ACE, Suit.CLUBS)),
+                            21
+                    ),
+
+                    new HandTestArgument(
+                            List.of(new Card(Rank.QUEEN, Suit.CLUBS),
+                                    new Card(Rank.EIGHT, Suit.CLUBS)),
+                            18
+                    ),
+
+                    new HandTestArgument(
+                            List.of(new Card(Rank.QUEEN, Suit.CLUBS),
+                                    new Card(Rank.ACE, Suit.CLUBS)),
+                            21
+                    ),
+
+                    new HandTestArgument(
+                            List.of(new Card(Rank.KING, Suit.CLUBS),
+                                    new Card(Rank.KING, Suit.CLUBS),
+                                    new Card(Rank.KING, Suit.CLUBS),
+                                    new Card(Rank.KING, Suit.CLUBS),
+                                    new Card(Rank.KING, Suit.CLUBS)),
+                            0
+                    ),
+
+                    new HandTestArgument(
+                            List.of(new Card(Rank.ACE, Suit.CLUBS),
+                                    new Card(Rank.ACE, Suit.CLUBS),
+                                    new Card(Rank.ACE, Suit.CLUBS),
+                                    new Card(Rank.KING, Suit.CLUBS),
+                                    new Card(Rank.KING, Suit.CLUBS)),
+                            0
+                    ),
+
+                    new HandTestArgument(
+                            List.of(new Card(Rank.ACE, Suit.CLUBS),
+                                    new Card(Rank.ACE, Suit.CLUBS),
+                                    new Card(Rank.NINE, Suit.CLUBS)),
+                            21
+                    )
+            );
+
+    /**
+     * Cards to test.
+     */
+    @DataPoints("objects")
+    public static final Set<Card> CARDS = Cards.getStandardDeck();
+
+
+    /**
+     * Tests that {@link Hand#withAdditionalCard(Card)} maintains the correct score.
+     *
+     * @param handTest
+     */
+    @Theory
+    @SuppressWarnings("checkstyle:methodname")
+    public void addCard_maintainsCorrectScore(HandTestArgument handTest) {
+        var hand = new Hand(handTest.getCards().get(0), handTest.getCards().get(1));
+        for (int i = 2; i < handTest.getCards().size(); i++) {
+            hand = hand.withAdditionalCard(handTest.getCards().get(i));
+        }
+        assertThat("The best score must match the target", hand.bestScore(), is(handTest.getTargetScore()));
+    }
+
+    private static final class HandTestArgument {
```

Comment:
Don't mix classes and methods. Either put this class at the top, or put it at the bottom.